rough-ideas.md

```diff
@@ -5,14 +5,15 @@ These are a rough set of fields in the `.bimg` format.
 A line is a tuple of (text, text colors, background colors) in blit format.  
 Frames are lists of lines.  
 Image files may be a single frame, or list of frames.  
+Each frame can have a `palette` key to specify a specific palette for that frame, following the rules of the base `palette` key.
   
 # Metadata
 Metadata is stored at the root of the table.    
 `version`: string Version string of the format.  
 `animation`: boolean Whether the image is an animation (if so, each frame is numerically indexed).  
 (per frame) `duration`: number Duration of the frame in seconds, overrides `secondsPerFrame` for that frame.  
 `secondsPerFrame`: number Default length of each frame in seconds.  
-`palette`: table Table of color tables indexed by frame. Each color table is in format of colors api (key: color number - value: EITHER table of 3 ranged 0-1 floats for each RGB component, OR a single item which is the hex-formatted number #RRGGBB)  
+`palette`: table Palette table in format of colors api (key: color number - value: EITHER table of 3 ranged 0-1 floats for each RGB component, OR a single item which is the hex-formatted number #RRGGBB). This is optional and will serve as the starting palette for any animations or images.
```

Comment:
The optional statement isn't required, as every single one of these metadata flags (save `animation`) are required.